src/core/tsi/fake_transport_security.c

```diff
@@ -74,6 +74,14 @@ typedef struct {
   size_t max_frame_size;
 } tsi_fake_frame_protector;
 
+typedef struct {
+  tsi_zero_copy_grpc_protector base;
+  grpc_slice_buffer *header_sb;
```

Comment:
Why are these two fields pointers?  It seems like we could save a couple of allocations by simply declaring them as direct members.